server/src/main/java/org/apache/calcite/avatica/server/HttpServer.java

```diff
@@ -337,17 +343,23 @@ protected ConstraintSecurityHandler configureSpnego(Server server,
       AvaticaServerConfiguration config) {
     final String realm = Objects.requireNonNull(config.getKerberosRealm());
     final String principal = Objects.requireNonNull(config.getKerberosPrincipal());
-
-    // A customization of SpnegoLoginService to explicitly set the server's principal, otherwise
-    // we would have to require a custom file to set the server's principal.
-    PropertyBasedSpnegoLoginService spnegoLoginService =
-        new PropertyBasedSpnegoLoginService(realm, principal);
+    final String principalWithoutRealm = principal.split("@", 2)[0];
+    final String[] principalParts = principalWithoutRealm.split("/", 2);
+    final String serviceName = principalParts[0];
+    final String hostname = principalParts[1];
+    final Path keytabPath = config.getKerberosKeytab();
+
+    ConfigurableSpnegoLoginService spnegoLoginService =
+        new ConfigurableSpnegoLoginService(realm, new TestAuthorizationService());
```

Comment:
`ConfigurableSpnegoLoginService` takes an `AuthorizationService` to configure roles. Not sure why this is necessary at all. This is one of the parts I am really confused on. I tried with `HashLoginService` but that didn't work OOTB.

Here is an example test:
https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-client/src/test/java/org/eclipse/jetty/client/util/SPNEGOAuthenticationTest.java#L141